### DISCUSSION TITLE
Converting WMF to PNG very slow ??

### DISCUSSION THREAD

**Discussion by Dantoys** [ImageMagick/ImageMagick#6449] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10 x64

### Description

Hello, I have images that are black and white in the WMF file format. Because of software changes they all need to be in the PNG format now. I am glad I found imagemagick, and through some bleep bloop searches and trial and order, I found this command that seems to do a great job and works in the new software. However the number of files are in the 5000+ range, and future projects could see 10,000 to 50,000 files needing converted. After 24hrs the process is still running. Please let me know if anyone cn help tweak the below command to speed up the conversion. Thank -you in advance..

mogrify -colors 2 -alpha Transparent -monitor -strokewidth 3 -format png *.wmf

**Comment by snibgo** [comment]
What is taking the most time? Perhaps "`-colors 2`". Do you really need that? Would a much simpler "`-threshold`" do what you want?

Can you link to a sample WMF file?

Also: does your command use multi-threading? GPU?

**Reply by Dantoys** [reply]
Yes, definitely the -colors 2 option. it runs through a dithering processes with each image. however if I remove this option, the image is a bit larger and it does not display correctly. I will try the -Threshold.. I am new to imagemagick, Thank-you. I will update after the testing. I assume the command prompt uses what ever the mogrify command uses. Looking at the Task manager, the processes seems to run on 1 CPU. Is there a way to specify multiple CPU's in mogrify?

**Comment by snibgo** [comment]
> ... and it does not display correctly.

In what way is it not correct?

I suggest using `magick mogrify` instead of `mogrify`.

Please paste the text output from both "`magick -version`" and "`mogrify -version`".

If they include OpenMP, then you will get multithreading. If they include OpenCL, then the command could use GPU, but only if you run the command "`set MAGICK_OCL_DEVICE=true`" first.

**Reply by Dantoys** [reply]
Below is the version information. the option "set MAGICK_OCL_DEVICE=true" does not speed up the process.

If I leave out the option for -color 2 then it runs real fast, I end up with a blank image. It must convert everything to transparent. 

in the previous message you mentioned -threshold. however i didn't find much in the documentation. I will need to experiment.

here is a sample file. 

> 
> Version: ImageMagick 7.1.1-12 Q8 x64 a09d8dd:20230625 https://imagemagick.org
> Copyright: (C) 1999 ImageMagick Studio LLC
> License: https://imagemagick.org/script/license.php
> Features: Cipher DPC Modules OpenCL OpenMP(2.0)
> Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
> Compiler: Visual Studio 2022 (193632532)
> 
> D:\Database\signatures>mogrify -version
> Version: ImageMagick 7.1.1-12 Q8 x64 a09d8dd:20230625 https://imagemagick.org
> Copyright: (C) 1999 ImageMagick Studio LLC
> License: https://imagemagick.org/script/license.php
> Features: Cipher DPC Modules OpenCL OpenMP(2.0)
> Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png p
[signature.zip](https://github.com/ImageMagick/ImageMagick/files/11910021/signature.zip)
s raqm raw rsvg tiff webp xml zip zlib
> Compiler: Visual Studio 2022 (193632532)

**Comment by snibgo** [comment]
>  It must convert everything to transparent.

Yeah. "`-alpha Transparent`" makes all pixels transparent. I don't know why you do this.

Try removing "`-alpha Transparent`" and "`-colors 2`". Does that give the results you want?

You have both OpenCL and OpenMP. It may be faster if you also "`set MAGICK_OCL_DEVICE=true`" before the `magick` command. But it may make no difference, as you will no longer be doing any processing.

**Comment by Dantoys** [comment]
THANK-YOU @snibgo 

After experimenting, I removed the -color option, and changed the order of the command options, and now the conversion works and is much much faster. I appreciate your input... Definitely will keep experimenting with threshold and other options, for now this command line gets me where I want to go..

magick mogrify -monitor -format png *.wmf -alpha Transparent -strokewidth 3
